### ISSUE TITLE
Modelo de ML funcional

### ISSUE THREAD

**Issue by aitorph7** [AI-School-F5-P3/G1_ML_Airline#4]
Un modelo de ML funcional que prediga el grado de satisfacción de los clientes.

Debe cumplir con los siguientes requisitos:

- [x] Overfitting inferior al 5%

**Comment by aitorph7** [comment]
Hemos utilizado el **algoritmo de clasificación CatBoost**, obteniendo unos resultados de validación y testeo más que aceptables:
Hablamos de unos datos de accuracy del 96-97%
Gracias a que partimos de un dataset muy limpio y balanceado.

Por otro lado, como alternativa vamos a utilizar un **algortimo de clasificación Random Forest** para obtener un análisis de importancia de características del modelo, es decir, averiguar cuáles son las características que más influyen en la satisfacción del cliente.
Estos datos son importantes para realizar el informe de negocio.

**Comment by aitorph7** [comment]
Tenemos el modelo de ML plenamente funcional con algoritmo CatBoost.

Hemos realizado una triple comparación entre algoritmos de clasificación para valorar el **"Feature Importance"** que nos devueklve cada uno.
- CatBoost
- XGBoost
- Random Forest

Hemos elaborado un análisis comparativo entre las ventajas/ inconvenientes que aporta cada algoritmo.
De acuerdo con él hemos razonado nuestra elección del algoritmo, que incluiremos en la presentación de Negocio y en la presentación Técnica.
